pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1023,7 +1045,7 @@ func buildListener(obj config.Config, l k8s.Listener, listenerIndex int) (*istio
 			// TODO currently we 1:1 support protocols in the API. If this changes we may
 			// need more logic here.
 			Protocol: string(l.Protocol),
-			Name:     fmt.Sprintf("%v-%v-gateway-%s-%s", strings.ToLower(string(l.Protocol)), l.Port, obj.Name, obj.Namespace),
+			Name:     fmt.Sprintf("%d-gateway-%s-%s", listenerIndex, obj.Name, obj.Namespace),
```

Comment:
Seems a little less descriptive – the `host` -> `hostname` seems to give this mapping a more human-readable way. Any reason not to include both?

Reply:
This should be entirely internal, so I am not sure the name matters too much. I think hostname may not be distinct and may have some characters like `*` that could cause issues?

Reply:
I figured the purpose was debugging – in which case the index is enough.